server/service/src/campaign/upsert.rs

```diff
@@ -74,8 +74,11 @@ fn validate(
     input: &UpsertCampaign,
 ) -> Result<(), UpsertCampaignError> {
     // Check for duplicate name
-    let campaigns_with_duplicate_name = CampaignRepository::new(connection)
-        .query_by_filter(CampaignFilter::new().name(StringFilter::equal_to(input.name.trim())))?;
+    let campaigns_with_duplicate_name = CampaignRepository::new(connection).query_by_filter(
+        CampaignFilter::new()
+            .name(StringFilter::equal_to(input.name.trim()))
+            .id(EqualFilter::not_equal_to(&input.id)),
+    )?;
```

Comment:
This was causing a problem -- it would give an error when updating and exiting row if the name was the same, so I added additional clause to only give this error when *inserting* a new campaign, not updating an existing one.